blocklib/digitizers/signal_averager/signal_averager_cpu.cc

```diff
@@ -0,0 +1,75 @@
+#include "signal_averager_cpu.h"
+#include "signal_averager_cpu_gen.h"
+
+#include "tags.h"
+
+namespace gr::digitizers {
+
+signal_averager_cpu::signal_averager_cpu(block_args args)
+    : INHERITED_CONSTRUCTORS {
+    set_tag_propagation_policy(tag_propagation_policy_t::TPP_CUSTOM);
+    set_relative_rate(1. / args.window_size); // TODO(PORT) This doesn't seem to have any effect?
+}
+
+work_return_t signal_averager_cpu::work(work_io &wio) {
+    const auto decim         = pmtf::get_as<std::size_t>(*this->param_window_size);
+
+    auto       noutput_items = std::numeric_limits<std::size_t>::max();
+    for (auto &w : wio.inputs()) {
```

Comment:
In many places where we have `auto&`, it could be turned into `const auto&`

Reply:
Done